Velocity smoother ros2 port

clalancette opened the pull request:
A port of the velocity smoother to ROS 2.  A few limitations:

1.  Compile tested against Eloquent
1.  Tested to launch in a simple way
1.  Not tested as part of a full application
1.  Launch files not ported
1.  Tests not ported
1.  Dynamic changing of parameters not yet supported

This PR also adds empty COLCON_IGNORE files for all of the pieces that don't yet compile against ROS 2.

clalancette committed:
Add in a direct copy of the kobuki_velocity_smoother.

Originally copied from https://github.com/yujinrobot/yujin_ocs/tree/17337e5a2d0a0f3711c55e272e656eb59174d657/yocs_velocity_smoother

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Port kobuku_velocity_smoother to ROS 2.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
COLCON_IGNORE all of the pieces not yet ported.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

stonier commented on kobuki_velocity_smoother/include/kobuki_velocity_smoother/velocity_smoother.hpp:
```diff
@@ -0,0 +1,104 @@
+ * @brief Velocity smoother interface
+ *
+ * License: BSD
+ *   https://raw.github.com/yujinrobot/yujin_ocs/hydro/yocs_velocity_smoother/LICENSE
```
Licenses ... this repo generally handles it by copy pasting the content up front here. Should follow that.

clalancette replied:
Fixed in ad27a3a

clalancette committed:
yocs -> kobuki everywhere in the velocity_smoother.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

stonier reviewed (approved):
Pending license update, this looks good to go - albeit partial.

clalancette committed:
Make sure things are thread-safe.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Add in the ability to change parameters on-the-fly.

This also required us to do some more work on concurrency
issues around the parameters.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Add in BSD license headers as per review.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Port the launchfiles to ROS 2.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette commented:
I ended up doing some work here to make the parameters configurable on-the-fly, fixing up any threading issues, and porting the launchfiles to ROS 2.  @stonier I think this is good to go now, so feel free to merge at your convenience.

clalancette committed:
Remove workaround for ecl_core bug now that it has been fixed.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Make sure rclcpp_components is an exec_depend.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

stonier commented:
You know, might have been nice for the history to get the original code in, then the changes so we can do a diff across commits. Let's talk about that tomorrow, we can probably still easily achieve that.

clalancette commented:
> You know, might have been nice for the history to get the original code in, then the changes so we can do a diff across commits. Let's talk about that tomorrow, we can probably still easily achieve that.

The commit history is actually setup like that here.  Commit 76f068fd826713de23436bd9d3f34e31cfcdb5df is a direct copy of the original code, and then the follow-on commits actually do the port.

That all being said, after looking at this package, I'm almost thinking we should make it a separate package.  The functionality is really pretty generic, and not tied to kobuki at all, so we could easily do that.

clalancette committed:
Remove the thread and use a timer instead.

This is much simpler, uses more of the ROS infrastructure,
allows us to remove a dependency on ecl_threads, and also
allows us to remove a slew of locking.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Add _ to the end of all class variables.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

stonier commented:
Aye, I had those two thoughts (interactively rebase these commits & wish for a separate package) as I hit the hay last night but couldn't be bothered to immediately get back to the desk and post them!

Next steps:

* I'll shift over this package as is (sans this PR) to https://github.com/kobuki-base/kobuki_ros and COLCON_IGNORE everything bar the metapackage. Then I'll release and make sure we have webhooks working so we can incrementally unignore everything with confidence.
* New home for you to move the velocity smoother to at https://github.com/kobuki-base/velocity_smoother.

clalancette commented:
Closing in favor of https://github.com/kobuki-base/velocity_smoother/pull/1
